cdap-master/src/main/java/co/cask/cdap/data/runtime/main/MasterServiceMain.java

```diff
@@ -488,7 +489,12 @@ private void monitorTwillApplication(final ScheduledExecutorService executor, fi
     if (controller != null) {
       startTime = 0L;
     } else {
-      controller = startTwillApplication();
+      try {
+        controller = startTwillApplication();
+      } catch (Exception e) {
+        LOG.error("Failed to start master twill application", e);
```

Comment:
I found that Twill only logs exception here at some level lower than ERROR, so I added this.